File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l=0;
        int r=0;
        int len=0;
        int n=s.length();
        HashSet<Character> hash = new HashSet<>();
        while(r<n){
            while(hash.contains(s.charAt(r))!=false){
                hash.remove(s.charAt(l));
                l++;
            }
            hash.add(s.charAt(r));
            len=Math.max(len,r-l+1);
            r++;
        }
        return (len==Integer.MIN_VALUE)?0:len;

    }
     

}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l=0;
        int r=0;
        int n=s.size();
        int maxi=0;
        map<char,int> mpp;
        while(r<n){
            if(mpp.find(s[r])!=mpp.end()){
                if(l<mpp[s[r]]+1){
                    l=mpp[s[r]]+1;

                }  
            }
            int len=r-l+1;
                maxi=max(maxi,len);
            mpp[s[r]]=r;
            r++;

        }
        return maxi;
    }
};